front/src/Store.ts

```diff
@@ -20,13 +22,25 @@ const reduxDevToolsOptions: Config = {
   },
 };
 
+const buildListenerMiddleware = () => {
+  const listener = createListenerMiddleware();
+  registerSimulationUpdateInfraIDListeners(listener);
+  registerStdcmUpdateInfraIDListeners(listener);
+
+  return listener;
+};
```

Comment:
I think that switchTypes is only used in the editor, so maybe we should move the call to fetch the switch type directly in the editor, rather call it every time we change infra in the other applications. If we do this, we don't need this listener anymore
```suggestion
const buildListenerMiddleware = () => {
  const listener = createListenerMiddleware();
  registerSimulationUpdateInfraIDListeners(listener);
  registerStdcmUpdateInfraIDListeners(listener);

  return listener;
};
```